tests/README.md

```diff
@@ -0,0 +1,173 @@
+
+These test inputs exist alongside each pipeline in their respective `test_inputs` directory.
+
+#### Truth Branches
+
+The testing framework has a concept of "truth branches". This allows users to define a branch of truth output to run against.
+
+In most cases the branch being run against is "master" which corresponds to a google storage bucket containing the agreed upon output of the pipeline as it exists in the "master" git branch.
+
+#### Results vs Storage buckets
+
+Every workflow run in cromwell has its output stored in an "execution" bucket. When running the pipeline tests it's important that we store this ouptut in a location that we can later reference if needed.
+
+All test workflows that are run will have their output copied to a "results" bucket -> `gs://broad-gotc-test-results/`
+
+When we want to update the "truth" branches for our pipelines we can copy that output to the respective "truth" bucket -> `gs://broad-gotc-test-storage/`
+
+## :hammer_and_pick: Testing Process
+
+We use Jenkins to run our test suite with two webhooks to kick off tests: PR to dev branch kicks off plumbing tests, PR to main/master kicks off scientific tests.
+
+Jenkins will run a job called `Smart Tests` which can be viewed in the [dsp-jenkins](https://github.com/broadinstitute/dsp-jenkins/blob/master/jobs/gotc-jenkins/WarpSmartTestJob.groovy) repo.
+
+The job of the smart tests is to run the `get_changed_pipeline_worklow_test_args.sh` bash script. The script performs a git diff against the target branch and recursively searches for any pipelines that have been changed either directly or indirectly via an imported sub-workflow.
+
+For each pipeline that has changed it will prepare the arguments for the Scala tool that submits the wrapper workflow to cromwell.
+
+Finally, it passes these arguments to the downstream [Workflow Test](https://github.com/broadinstitute/dsp-jenkins/blob/master/jobs/gotc-jenkins/WarpWorkflowTestsJob.groovy) job which calls the Scala `CloudWorkflow` command and submits the wrapper workflow.
+
+### Scala Framework
+
+As mentioned above, the test framework leverages Scala as a CLI tool to submit the wrapper workflows to Cromwell. The code for the Scala tool can be found in the [tests/broad/scala_test](../tests/broad/scala_test/) directory.
+
+The Scala tool is responsible for preparing the inputs for the wrapper workflow of the given pipeline and submitting that workflow to the cromwell environment specified.
+
+**NOTE** - When looking at the the scala code you will see that there are multiple 'tester' files. A major rewrite of the framework was done to only use one tester command (CloudWorkflow) and to pass in the desired pipeline as an argument. This change makes it easier for contributors and maintainers of these workflows to write tests by pushing the functionality of the test framework away from Scala and into the wrapper workflows.
```

Comment:
Will these other tester files be cleaned up when we do the 'clean up old test infra' ticket?

Reply:
Not my call, you guys can decide what you would like to do with it, leave it there for posterity or rip it out. You will run the risk of breaking some things if you decide to remove things!